compiler/src/dotty/tools/dotc/reporting/diagnostic/ErrorCategory.scala

```diff
@@ -0,0 +1,20 @@
+package dotty.tools.dotc.reporting.diagnostic
+
+sealed case class ErrorCategory(name: String) {
+  override def toString: String = name
+}
+object ErrorCategories {
+  val COMPATIBILITY = ErrorCategory("Compatibility")
```

Comment:
Please use Scala naming convention: `NoKind` instead of `NO_KIND`. Also I would put `NoKind` at the top.

Reply:
I can definitely rename everything, but it unfortunately clashes with the `case class PatternMatchExhaustivity(uncovered: String)` (both the message and kind clash together). What would you recommend ? Keep the same name but prefix like ` val kind = ErrorKind.PatternMatchExhaustivity` or rename the enum members ? (if so, how ?). Thanks for your insights

Reply:
Yes you can disambiguate with `val kind = ErrorCategory.PatternMatchExhaustivity`